cpp/include/cudf/stream_compaction.hpp

```diff
@@ -101,6 +101,81 @@ std::unique_ptr<table> drop_nulls(
   std::vector<size_type> const& keys,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Filters a table to remove NANs with threshold count.
+ *
+ * Filters the rows of the `input` considering specified columns indicated in
+ * `keys` for NANs. These key columns must be of floating-point type.
+ *
+ * Given an input table_view, row `i` from the input columns is copied to
+ * the output if the same row `i` of @p keys has at least @p keep_threshold
+ * non-NAN elements.
+ *
+ * This operation is stable: the input order is preserved in the output.
+ *
+ * @code{.pseudo}
+ *          input   {col1: {1.0, 2.0, 3.0, NAN},
+ *                   col2: {4.0, null, NAN, NAN},
+ *                   col3: {7.0, NAN, NAN, NAN}}
+ *          keys = {0, 1, 2} // All columns
+ *          keep_threshold = 2
+ *
+ *          output {col1: {1.0, 2.0}
+ *                  col2: {4.0, null}
+ *                  col3: {7.0, NAN}}
+ * @endcode
+ *
+ * @note if @p input.num_rows() is zero, or @p keys is empty,
+ * there is no error, and an empty `table` is returned
+ *
+ * @throws cudf::logic_error if The `keys` columns are not floating-point type.
+ *
+ * @param[in] input The input `table_view` to filter.
+ * @param[in] keys  vector of indices representing key columns from `input`
+ * @param[in] keep_threshold The minimum number of non-NAN elements in a row
+ *                           required to keep the row.
+ * @param[in] mr Device memory resource used to allocate the returned table's device memory
+ * @return Table containing all rows of the `input` with at least @p
+ * keep_threshold non-NAN elements in @p keys.
+ */
+std::unique_ptr<table> drop_nans(
+  table_view const& input,
+  std::vector<size_type> const& keys,
+  cudf::size_type keep_threshold,
```

Comment:
would 1 be a reasonable/useful default here?

Reply:
1 works, and cudf by default uses 1 as well.

Reply:
in that case:
```suggestion
  cudf::size_type keep_threshold = 1,
```

Reply:
I just realized, two `drop_nans` just differ because of `keep_threshold` and if we make it default, then there will be conflict between both. So, can't have default value.

Reply:
So `keep_threshold` default is effectively the number of columns.
Is this fine, given that the Python API default is one?

Reply:
We have two libcudf `dropna`, one with `keep_threshold` and and one without which it. So, making this default would be an issue as compiler will have issues identifying the proper function. 

And my mistake, I got little confused with `keep_threshold` in libcudf and python. So, by default in a row, there shouldn't be a single `None` -> `keep_threshold=keys.size()` and not 1.